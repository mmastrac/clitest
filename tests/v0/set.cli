#!/usr/bin/env clitest --v0

set FOO bar;

$ echo $FOO
! bar

set A 1;
set B 2;

set FOO "$A $B";

$ echo $FOO
! 1 2

# The below lines match 'sh', not bash!
set VAR 'a\\b';
$ echo $VAR
if TARGET_OS != "windows" {
    ! a\b
}
if TARGET_OS == "windows" {
    ! a\\b
}

set VAR "a\\n";
$ echo $VAR
if TARGET_OS != "windows" {
    ! a
    !
}
if TARGET_OS == "windows" {
    ! a\n
}

set VAR "a\$b";
$ echo $VAR
! a$b

set VAR 'a\$b';
$ echo $VAR
! a\$b

set VAR "test";
if true {
    # This escapes the block
    set VAR "test2";
    $ echo $VAR
    ! test2
}

$ echo $VAR
! test2
