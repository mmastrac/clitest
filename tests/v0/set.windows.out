Running <root>/set.cli ...

set FOO bar

echo $FOO
---
bar
---
✅ OK

set A 1

set B 2

set FOO 1 2

echo $FOO
---
1 2
---
✅ OK

set VAR a\\b

echo $VAR
---
a\\b
---
✅ OK

set VAR a\n

echo $VAR
---
a\n
---
✅ OK

set VAR a$b

echo $VAR
---
a$b
---
✅ OK

set VAR a\$b

echo $VAR
---
a\$b
---
✅ OK

set VAR test

set VAR test2

echo $VAR
---
test2
---
✅ OK

echo $VAR
---
test2
---
✅ OK

<root>/set.cli PASSED
